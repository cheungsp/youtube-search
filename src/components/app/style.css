body {
  background-color: whitesmoke;
}

.video-item img {
  max-width: 64px;
}

.video-detail .details {
  margin-top: 10px;
  padding: 10px;
  border: 1px solid #ddd;
  border-radius: 4px;
}