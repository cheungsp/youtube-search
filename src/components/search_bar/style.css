.search-bar {
  margin: 20px;
  text-align: center;
}

.search-bar input {
  width: 80%;
}

/*
"test": "jest"
"test": "react-scripts test --env=jsdom",
*/