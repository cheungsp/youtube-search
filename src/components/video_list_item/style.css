.single-video-item {
  margin-bottom: 15px;
}