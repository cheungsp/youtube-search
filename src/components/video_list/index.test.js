import React from 'react';
import { shallow } from 'enzyme'
import VideoList from './index.js'
import VideoListItem from '../video_list_item/index'
import toJson from 'enzyme-to-json'
import './style.css';


test('VideoList component should render as expected', () => {
  const component = shallow(<VideoList videos={[]}/>)
  const tree =  toJson(component)
  expect(component.find(VideoListItem).length).toBe(0);
  expect(tree).toMatchSnapshot()
})

test('VideoList component should render as expected', () => {
  const component = shallow(<VideoList videos={[
    {
      'title': 'video 1',
      'key': 1,
      'id': 1
    },
    {
      'title': 'video 2',
      'key': 2,
      'id': 2
    }
  ]}/>)
  const tree =  toJson(component)
  expect(component.find(VideoListItem).length).toBe(2);
  expect(tree).toMatchSnapshot()
})
