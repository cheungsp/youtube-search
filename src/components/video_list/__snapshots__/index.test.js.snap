// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList component should render as expected 1`] = `
<ul
  className="col-md-4"
/>
`;

exports[`VideoList component should render as expected 2`] = `
<ul
  className="col-md-4"
>
  <VideoListItem
    video={
      Object {
        "id": 1,
        "key": 1,
        "title": "video 1",
      }
    }
  />
  <VideoListItem
    video={
      Object {
        "id": 2,
        "key": 2,
        "title": "video 2",
      }
    }
  />
</ul>
`;
